### PR TITLE
QT: remove deprecations for QT 5.15

### PR DESCRIPTION
**Pull request by dyfer**
## Purpose and Motivation

This PR provides updates and fixes for the deprecations up to Qt 5.15.
This is an "intermediary" PR, extracted from the Qt6 update branch for easier review. It generally does not include Qt6 code explicitly, even though commits do mention qt6...

Thanks @jrsurge for starting this work back in the day! A number of commits were extracted from @jrsurge's qt6 branch.

This PR also raises the minimum supported QT to 5.15; however I'd wait with updating documentation until the Qt6 updates are finalized.

Please review carefully! I'm not confident in C++ nor in Qt :)

## Types of changes

- Bug fix (?)

## To-do list

- [x] Code is tested
- [x] All tests are passing
- n/a Updated documentation
- [x] This PR is ready for review

### CODE REVIEW

**Comment by dyfer** [comment]
Anybody interested in reviewing this? @JordanHendersonMusic @capital-G @telephon ? 😀

**Review by JordanHendersonMusic — APPROVED** [review]
This seems like a very clear one to one translation.

I did wonder about load and load relaxed, but the docs say load was relaxed.

Bring on qt6!!

File: editors/sc-ide/widgets/multi_editor.hpp [hunk 1]
```diff
@@ -32,6 +32,8 @@
 #include <QTextDocument>
 #include <QSplitter>
 
+#include "../doc_manager.hpp"
+
```
Comment (capital-G) [inline editors/sc-ide/widgets/multi_editor.hpp:36]: just out of curiosity - how do the occasional additional new imports are coming into place here? Are they Qt specific, do they help the compiler/IDE to be more explicit or are these just left-overs from development?

**Review by capital-G — APPROVED** [review]
Thanks so much for this step towards Qt6 :)

File: editors/sc-ide/widgets/multi_editor.hpp (hunk 1 above)
Comment (capital-G) [reply editors/sc-ide/widgets/multi_editor.hpp:36]: Interesting. Looking forward to the Qt6 PR :)

Comment (dyfer) [reply editors/sc-ide/widgets/multi_editor.hpp:36]: Coming up! Once the previous CI passes, so that we have that brew universal cache...

**Pull request merged**